### PR TITLE
Package Datadog Agent for DCOS to monitor mesos

### PR DESCRIPTION
**Pull request by alq666**
[Datadog](http://datadog.com) can monitor mesos and marathon natively (this version only support mesos). It uses an agent that runs as a docker container and queries the stats endpoints.

Here is what it looks like when running on DCOS:

screenshot 2015-08-18 16 54 14

### CODE REVIEW

**Comment by alq666** [comment]
The conflict stems from `index.json.gz` being checked in, by the way.

File: repo/packages/D/datadog/0/marathon.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                {
+                    "key": "name", "value": "dd-agent"
+                },
+                {
+                    "key": "env", "value": "API_KEY={{datadog.api_key}}"
+                }
+            ],
+            "network": "BRIDGE",
+            "portMappings": [
+                { "hostPort": 8125, "containerPort": 8125, "protocol": "udp" },
```
Comment (ConnorDoyle) [inline repo/packages/D/datadog/0/marathon.json]: Could you please make the `hostPort` configurable for both of these?  Maybe you could default to the canonical values.

File: repo/packages/D/datadog/0/marathon.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
+    "id": "datadog-agent",
```
Comment (kensipe) [inline repo/packages/D/datadog/0/marathon.json]: we need this to be configurable... it is commonly framework-name, however this doesn't seem to be a framework.  We still need this to be a configurable {{data-dog.app-id}}

File: repo/packages/D/datadog/0/marathon.json (hunk 1 above)
Comment (alq666) [inline repo/packages/D/datadog/0/marathon.json]: Sure thing.

File: repo/packages/D/datadog/0/marathon.json (hunk 2 above)
Comment (alq666) [inline repo/packages/D/datadog/0/marathon.json]: Will do.

File: repo/packages/D/datadog/0/marathon.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
+{
+    "id": "{{datadog.app_id}}",
```
Comment (alq666) [inline repo/packages/D/datadog/0/marathon.json]: @kensipe does that work as a configurable?

Comment (kensipe) [inline repo/packages/D/datadog/0/marathon.json]: perfect! 
got busy with mesoscon etc... but I'm on this now.

**Comment by kensipe** [comment]
@alq666 I'm checking this out... the installation fails due to licensing with the following message:

```
Registered docker executor on ip-10-0-0-206.us-west-2.compute.internal
Starting task datadog-agent.e2b9de24-4c04-11e5-87d3-521641bcf454
You must set API_KEY environment variable to run the Datadog Agent container
```

1. can you help me out here
2. is there a place where this is documented?

**Comment by kensipe** [comment]
@alq666 
is it expected that a user must create a json file to over these values at the time of installation?

```
Pass your API key as a package option (datadog.api_key).
Also pass the number of Mesos slaves as an option (datadog.instances).
```

**Comment by alq666** [comment]
@kensipe it is so I'm curious if there's a better way to do this.

The API key is an obvious one. The number of instances is until we can run as an admin task (and it gets deployed on all the agent nodes).

**Comment by kensipe** [comment]
Tested... LGTM.  I'm running details past our internal team.

**Comment by kensipe** [comment]
@alq666 here is what I would like to do... will you remove the default value for api_key in config.json?  By removing it,  the DCOS CLI will see that it is required but that it is missing... this will fail the installation until a key is put in a json file and included with the `--options` flag.     This will be a better experience IMO.   After that... I'm inclined to merge this.  nice work!

**Comment by kensipe** [comment]
a bit usually... but I've tested the change I requested... so I will merge manual with the change and add this in the universe.

**Comment by alq666** [comment]
@kensipe thank you! :+1:

**Pull request merged**